components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt

```diff
@@ -153,15 +153,17 @@ class CrashReporter(
 
         logger.info("Received crash: $crash")
 
-        if (shouldSendIntent(crash)) {
-            // This crash was not fatal and the app has registered a pending intent
-            launchCrashIntent(context, crash)
-        }
-
         if (telemetryServices.isNotEmpty()) {
             sendCrashTelemetry(context, crash)
         }
 
+        // If crash is native code and non fatal then the view will handle the user prompt
+        if (crash is Crash.NativeCodeCrash && !crash.isFatal) {
+            // App has registered a pending intent
+            sendNonFatalCrashIntent(context, crash)
+            return
+        }
```

Comment:
Why do we switch to always doing this? Previously we did this only when we had an intent and that kind of made sense? Currently we explicit allow apps to not provide an Intent if they do not intent to handle crashes in their UI. 🤔

Reply:
> Why do we switch to always doing this? Previously we did this only when we had an intent and that kind of made sense? Currently we explicit allow apps to not provide an Intent if they do not intent to handle crashes in their UI. 🤔

You're right.  I'll update so if there's no intent it should continue on to report the crash.